src/backend/cuda/nvrtc/cache.cpp

```diff
@@ -149,6 +157,107 @@ void Kernel::getScalar(T &out, const char *name) {
 template void Kernel::setScalar<int>(const char *, int);
 template void Kernel::getScalar<int>(int &, const char *);
 
+#if !defined(OS_WIN)
+string getHomeDirectory() {
+    string home = getEnvVar("XDG_CONFIG_HOME");
+    if (!home.empty()) return home;
+
+    home = getEnvVar("HOME");
+    if (!home.empty()) return home;
+
+    home = getpwuid(getuid())->pw_dir;
+}
+#else
+string getTemporaryDirectory() {
+    DWORD bufSize = 261;  // limit according to GetTempPath documentation
+    string retVal;
+    retVal.resize(bufSize);
+    bufSize = GetTempPathA(bufSize, &retVal[0]);
+    retVal.resize(bufSize);
+    return retVal;
+}
+#endif
+
+bool folderExists(const string &path) {
+#if !defined(OS_WIN)
+    struct stat status;
+    return stat(path.c_str(), &status) == 0 && (status.st_mode & S_IFDIR) != 0;
+#else
+    struct _stat status;
+    return _stat(path.c_str(), &status) == 0 && (status.st_mode & S_IFDIR) != 0;
+#endif
+}
+
+bool createFolder(const string &path) {
+#if !defined(OS_WIN)
+    return mkdir(path.c_str(), 0777) == 0;
+#else
+    return CreateDirectoryA(path.c_str(), NULL) != 0;
+#endif
+}
+
+bool removeFile(const string &path) {
+#if !defined(OS_WIN)
+    return unlink(path.c_str()) == 0;
+#else
+    return DeleteFileA(path.c_str()) != 0;
+#endif
+}
+
+bool isDirectoryWritable(const string &path) {
+    if (!folderExists(path) && !createFolder(path)) return false;
+
+    const string testPath = path + AF_PATH_SEPARATOR + "test";
+    if (!std::ofstream(testPath).is_open()) return false;
+    removeFile(testPath);
+    
+    return true;
+}
+
```

Comment:
All the helper functions can go into `src/backend/common` since they are not specific to compute backends.

Reply:
Will do